docs: procedures related to GitHub teams and CODEOWNERS (#4173)
nl-design-system/documentatie: docs/handboek/developer/05-infrastructuur/03-github-teams-en-codeowners.md

---
title: GitHub teams en code owners
hide_title: true
hide_table_of_contents: false
sidebar_label: GitHub teams en code owners
pagination_label: GitHub teams en code owners
description: Het werken met GitHub teams en code owners binnen NL Design System
slug: /handboek/developer/github-teams-en-code-owners/
keywords:
  - GitHub
  - Teams
  - Codeowners
  - Code owners
---

# GitHub teams en code owners

> Binnen NL Design System gebruiken we GitHub teams om verantwoordelijkheden duidelijk te organiseren.

Reviewer: ```suggestion
Binnen NL Design System gebruiken we rollen om verantwoordelijkheden duidelijk te organiseren.
In GitHub zijn er GitHub Teams per organisatie en per rol.
```

Author: 👍 ik hou GitHub team lowercase T, volgens mij doen ze dat zelf ook zo (en met "GitHub Team" gerefereerd naar hun paid plan)

> Hiermee organiseren we wie (schrijf)toegang heeft tot repositories, en weet je snel wie je moet betrekken bij een wijziging, code review of toegangsvraag.

Reviewer: Het gebruik van haakjes in "(schrijf)toegang" is niet toegestaan volgens de schrijfwijzer

Reviewer 2: ```suggestion
Hiermee organiseren we wie toegang heeft tot repositories, en weet je snel wie je moet betrekken bij een wijziging, code review of toegangsvraag.
```

> Hiermee organiseren we wie (schrijf)toegang heeft tot repositories, en weet je snel wie je moet betrekken bij een wijziging, code review of toegangsvraag.

Reviewer: ```suggestion
Hiermee organiseren we wie toegang heeft tot repositories, en weet je snel wie toestemming moet geven voor een aanpassing van code of voor een wijziging in toegangsrechten.
```

> We gebruiken het `CODEOWNERS`-bestand om reviewverzoeken automatisch bij de juiste mensen te laten landen.

Reviewer: "landen" is beeldspraak

```suggestion
We gebruiken het `CODEOWNERS`-bestand om te garanderen dat een code review wordt gevraagd aan de juiste personen, voor wijzigingen aan belangrijke bestanden.
```

> ## Teams in de praktijk

Reviewer: ```suggestion
## Rollen
```

> Over het algemeen hanteren we per organisatie een contributor-team, een maintainer-team en een admin-team.

Reviewer: ```suggestion
Over het algemeen is er per organisatie een contributor-team, een maintainer-team en een admin-team.
```

- `<organisatie>-contributor` - heeft schrijftoegang tot de repository en kan daarmee pull requests indienen
- `<organisatie>-maintainer` - voor het beoordelen en mergen van pull requests en onderhoudstaken op de repository
- `<organisatie>-admin` - voor toegangsbeheer van de repository

> Op verzoek kunnen voor meer granulariteit meer teams worden aangemaakt, maar bovenstaande is in de meeste gevallen voldoende.

Reviewer: ```suggestion
Op verzoek kunnen extra rollen gebruikt worden, maar de standaard-rollen zijn in de meeste gevallen voldoende.
```

## Lid worden van een team

> Een maintainer of een lid van het kernteam kan in de `terraform` infrastructuurrepository een pull request aanmaken.

Reviewer: ```suggestion
Wanneer iemand aan een team toegevoegd moet worden, dan moet een maintainer of een admin die wijziging maken door een Pull Request te maken in de `terraform` repository.
```

Hiervoor is een [handleiding](https://github.com/nl-design-system/terraform/blob/main/docs/adding-users.md) beschikbaar.

> De admin van de organisatie wordt gevraagd om de aanvraag goed te keuren, en om te kijken of de lijst van gebruikers in de teams up-to-date is.

Reviewer: ```suggestion
De admin van de organisatie moet wijziging indienen of goedkeuren.

De admin moet periodiek controleren dat de toegangsrechten van gebruikers nog klopt.
```

## CODEOWNERS

> Met `CODEOWNERS` leg je vast welke gebruikers/teams eigenaar zijn van bepaalde bestanden of mappen.

Reviewer: ```suggestion
Met `CODEOWNERS` leg je vast welke gebruikers of rollen verantwoordelijk zijn voor bepaalde bestanden of mappen.
```

> Deze gebruikers worden automatisch toegevoegd als reviewer als een pull request deze bestanden of mappen wijzigt.

Reviewer: Misschien nog even het doel beschrijven?

Doordat ook niet blijkt dat die review verplicht is (voelt nu erg vrijblijvend) vraag ik me af of het kwartje valt dat dit bijvoorbeeld een beveiligingsmaatregel kan zijn

Een vereenvoudigd voorbeeld:

```txt
# Infra-bestanden

> .github/ @nl-design-system/kernteam-maintainer

Reviewer: Moeten we in het algemeen adviseren om de rollen met meer rechten achteraan te zetten? Dan moet `maintainer` ook na `@nl-design-system/documentatie` komen

docs/ @nl-design-system/documentatie
```

> Met dit voorbeeld, bij een pull request dat bestanden onder `docs/` wijzigt, vraagt GitHub automatisch een review aan bij `@nl-design-system/documentatie`.

Reviewer: ```suggestion
Met dit voorbeeld, bij een pull request dat bestanden onder `docs/` wijzigt, vraagt GitHub automatisch een review aan bij de gebruikers in het team `@nl-design-system/documentatie`.
```

:::tip[Praktisch advies]
Het kan zijn dat een team niet 1-op-1 mapt op de mensen die je als eigenaar wil toewijzen.
Gebruik in `CODEOWNERS` dan bij voorkeur individuele accounts in plaats van teams.

Gebruik verder comments bij paden waar uit de naamgeving niet direct duidelijk wordt wat de scope is van de bestanden.
:::

### Review van code owners verplichten

Code owners worden altijd aan pull requests toegevoegd als reviewer.

> Middels een _branch protection rule_ kun je verplichten dat een goedkeuring van één (of meer) code owners nodig is.

Reviewer: onnodige haakjes

```suggestion
Middels een _branch protection rule_ kun je verplichten dat een goedkeuring van één of meer code owners nodig is.
```
